File: app/src/main/java/com/mikhailovskii/trakttv/ui/base/BasePresenter.java
package com.mikhailovskii.trakttv.ui.base;

import androidx.annotation.NonNull;

import io.reactivex.disposables.CompositeDisposable;

public class BasePresenter<View> implements MvpPresenter<View> {

    public View mView = null;
    public CompositeDisposable mCompositeDisposable = new CompositeDisposable();

    @Override
    public void attachView(@NonNull View view) {
        mView = view;
    }

    @Override
    public void detachView() {
        mCompositeDisposable.clear();
        mView = null;
    }

}

File: app/src/main/java/com/mikhailovskii/trakttv/ui/base/BasePresenter.kt
package com.mikhailovskii.trakttv.ui.base

import io.reactivex.disposables.CompositeDisposable

open class BasePresenter<View> : MvpPresenter<View> {

    var view: View? = null
    var compositeDisposable = CompositeDisposable()

    override fun attachView(view: View) {
        this.view = view
    }

    override fun detachView() {
        compositeDisposable.clear()
        view = null
    }

}
